qt/applications/workbench/workbench/plugins/workspacewidget.py

```diff
@@ -203,13 +203,17 @@ def _run_create_detector_table(self, ws):
         CreateDetectorTable(InputWorkspace=ws)
 
     def _action_double_click_workspace(self, name):
+        ws = self._ads.retrieve(name)
         try:
             # if this is a table workspace (or peaks workspace),
             # then it can't be plotted automatically, so the data is shown instead
-            TableWorkspaceDisplay.supports(self._ads.retrieve(name))
+            TableWorkspaceDisplay.supports(ws)
             self._do_show_data([name])
         except ValueError:
-            plot_from_names([name], errors=False, overplot=False, show_colorfill_btn=True)
+            if ws.blocksize() > 1:
+                plot_from_names([name], errors=False, overplot=False, show_colorfill_btn=True)
+            else:
+                self._do_show_data([name])
```

Comment:
Sorry this is very pedantic but I think the function would look cleaner if it was written like this:
```
ws = self._ads.retrieve(name)
try:
    # if this is a table workspace (or peaks workspace),
    # then it can't be plotted automatically, so the data is shown instead
    TableWorkspaceDisplay.supports(ws)
except ValueError:
    if ws.blocksize() > 1:
        plot_from_names([name], errors=False, overplot=False, show_colorfill_btn=True)
        return

self._do_show_data([name])
```
It would mean that `_do_show_data` is only called in one place.

Reply:
Personally I dislike the return in the catch I think it makes the execution logic a bit too jumpy.

Reply:
Fair enough, this is good to go in that case 👍